### ISSUE TITLE
OHLCV data challenge with mplfinance

### ISSUE THREAD

**Issue by manniokim** [matplotlib/mplfinance#422]
Ask anything you want about mplfinance usage, project philosophy and/or priorities, or anything else related to mplfinance.

hello, I am new to data processing with pandas and python programming generally, I am working with a large crypto ohlcv dataset from a database. I need to answer some questions from the dataset, questions like 

1)which day of the week shows the greatest change in price?
2)which hour of the day shows the greatest change in price?
3)looking at weekly data, to what extent can the Thursday - Sunday price change be predicted based on the movements from Monday - Wednesday?
4)looking at weekly data, does the weekday pricing data provide any predictive clues to what will happen by the end of the weekend?

so far, I've been able to extract the data, set timestamp as index and tried to resample the data,plot the data in mplfinance but I have two challenges, I can't present the data as days of the week eg. Thursday and the picture from the plot doesn't answer my questions.is there a way for me to aswer this questions using the timestamp as it is?  I tried the alternative of creating a new data frame with just the variables I need (open, high,low,close,volume,date, time, day of the week) but this data won't plot in mplfinance. I would appreciate any help with this issue
<img width="1440" alt="Screenshot 2021-07-23 at 08 34 19" src="https://user-images.githubusercontent.com/81591365/126744009-c78f41ba-2faa-4fc3-ae81-1a2858450c75.png">
<img width="1440" alt="Screenshot 2021-07-23 at 08 34 34" src="https://user-images.githubusercontent.com/81591365/126744052-54f3202d-f19f-418e-8d51-b5c90b50a2fa.png">
<img width="1440" alt="Screenshot 2021-07-23 at 08 34 47" src="https://user-images.githubusercontent.com/81591365/126744056-73a3f847-67ed-4a5f-8968-ace5db5abe45.png">
<img width="1440" alt="Screenshot 2021-07-23 at 08 36 39" src="https://user-images.githubusercontent.com/81591365/126744061-e0a0e153-cc58-442f-b315-d41e76175a9f.png">
<img width="1440" alt="Screenshot 2021-07-23 at 08 37 07" src="https://user-images.githubusercontent.com/81591365/126744065-72d5ecbe-e7b8-4cf4-8c1e-eef7de383a8c.png">
<img width="1440" alt="Screenshot 2021-07-23 at 08 37 30" src="https://user-images.githubusercontent.com/81591365/126744096-273497bd-2e46-4e84-a875-f1ca2b7797b5.png">
<img width="859" alt="Screenshot 2021-07-23 at 08 37 50" src="https://user-images.githubusercontent.com/81591365/126744098-c7bce001-b20f-4ffa-98d2-bda4328a4c38.png">
<img width="1017" alt="Screenshot 2021-07-23 at 08 40 22" src="https://user-images.githubusercontent.com/81591365/126744099-0bf3b7db-71c7-41fd-b3a4-c9fd459b62ba.png">
<img width="981" alt="Screenshot 2021-07-23 at 08 59 43" src="https://user-images.githubusercontent.com/81591365/126744101-807d7f57-7665-4e03-9d11-ec369eafecf4.png">
<img width="1062" alt="Screenshot 2021-07-23 at 09 01 02" src="https://user-images.githubusercontent.com/81591365/126744102-c636411f-d927-4f28-a183-1863fcee2f35.png">
<img width="1440" alt="Screenshot 2021-07-23 at 09 04 59" src="https://user-images.githubusercontent.com/81591365/126744103-7a0994cf-11bb-45df-abfb-3b5c6247a55f.png">
<img width="1440" alt="Screenshot 2021-07-23 at 09 05 05" src="https://user-images.githubusercontent.com/81591365/126744104-8f8c8a46-4ab0-4bd9-b9aa-b1cb228d8419.png">
<img width="1440" alt="Screenshot 2021-07-23 at 09 05 10" src="https://user-images.githubusercontent.com/81591365/126744107-85a2ca6a-1384-434d-bfee-f397744ee7b0.png">
. Thank you

**Comment by DanielGoldfarb** [comment]
@manniokim 
Hi.  Thank you for contributing your issue to mplfinance.   I have some idea what you are trying to accomplish, but it is not entirely clear to me.  Also, while helpful, it is very tedious to go through in detail all of the images of your notebook.

Would you be willing to post your notebook, as it is in these pictures (that is, with both inputs ***and outputs***) to a github repository or gist and give me access to it?   That way I can possibly clone it and see if I can solve the issue you are having.

Thanks again.   All the best.   --Daniel

**Comment by manniokim** [comment]
> @manniokim
> Hi. Thank you for contributing your issue to mplfinance. I have some idea what you are trying to accomplish, but it is not entirely clear to me. Also, while helpful, it is very tedious to go through in detail all of the images of your notebook.
> 
> Would you be willing to post your notebook, as it is in these pictures (that is, with both inputs _**and outputs**_) to a github repository or gist and give me access to it? That way I can possibly clone it and see if I can solve the issue you are having.
> 
> Thanks again. All the best. --Daniel

**Comment by manniokim** [comment]
@DanielGoldfarb I have posted the notebook and given you access to it. thank you.

**Comment by DanielGoldfarb** [comment]
@manniokim 
Thanks.  I have access to the repository (ohlcvdata) but I don't yet see the notebook there.

**Comment by DanielGoldfarb** [comment]
@manniokim 
Thanks.  I see the notebook now.  I will try to take a look in the next day or two.   All the best.  --D.

**Comment by DanielGoldfarb** [comment]
@manniokim  ... Can you please add please add the file **`list.csv`** to the repository?   Thanks.

**Comment by manniokim** [comment]
@DanielGoldfarb done, thank you

**Comment by manniokim** [comment]
Hi @DanielGoldfarb  checking in

**Comment by DanielGoldfarb** [comment]
@manniokim  
I still don't see `list.csv` in [the repository](https://github.com/manniokim/ohlcvdata/blob/main/OHLCV%20data.ipynb).   I can see the results of it in the notebook, which is helpful.  And I believe that I can give you some ideas as to how to answer your questions.  But I think I may be able to give you a  more complete answer if I run your notebook and play around with the data some.  However if I clone the repository and run your notebook, I think it will fail on `pd.read_csv('list.csv')` because I don't see the csv file in the repository.  (You need to `git add list.csv` and then `git commit`).

I am hoping later today (a few hours from now) to have some time to look into this and respond.

**Comment by manniokim** [comment]
@DanielGoldfarb I'm sorry for the oversight. I have now added it

**Comment by manniokim** [comment]
@DanielGoldfarb checking in

**Comment by DanielGoldfarb** [comment]
I have the csv file, everything I need ... I just haven't had time with other projects.  Hope to look at this early next week.

**Comment by manniokim** [comment]
Thank you

**Comment by manniokim** [comment]
@DanielGoldfarb Hi, I was able to find way around using Vlines. I was also able to answer the questions , however I still need help answering the question with the data as an aggregate not singly. I appreciate if you can spare some time to guide me.

**Comment by manniokim** [comment]
@DanielGoldfarb I have invited you as collaborator on the new repo

**Comment by DanielGoldfarb** [comment]
> @DanielGoldfarb I have invited you as collaborator on the new repo

@manniokim Okim, very nice job on the notebook, very organized and clear.

---

> @DanielGoldfarb Hi, I was able to find way around using Vlines. I was also able to answer the questions , however I still need help answering the question with the data as an aggregate not singly. I appreciate if you can spare some time to guide me.

So just to clarify what you mean by "answering the question with the data as an aggregate not singly" ... I see, for example, your first question "Which day of the week shows the greatest change in price?" you have answered "The day that shows the greatest change in price is the 9th of June which is a Wednesday with an opening price of 37500 and a closing price of 36,200."  

Based on this, I would assume by "as an aggregate not singly" you mean that you want to compare all Mondays in aggregate, to all Tuesdays in aggregate, to all Wednesdays, in aggregate, etc.   **Is this correct?**  

If so, then the question becomes, *how best to aggregate them to make sense of the answer?*  For example, we see that one particular Wednesday has the absolute largest change, but maybe there are two or three Tuesdays that have then 2nd, 3rd, and 4th largest changes.  If that were the case, then even though one Wednesday has the largest change of all, we see that large changes are more common on Tuesdays.  *Is that the type of "aggregate" question you are trying to answer?*

**Comment by manniokim** [comment]
@DanielGoldfarb yes please, that is the type of aggregate question I'm trying to answer. eg. is it more profitable to trade on say Tuesdays than on Thursdays ? apologies for not  framing the question clearly.

**Comment by manniokim** [comment]
@DanielGoldfarb Hi, checking in

**Comment by DanielGoldfarb** [comment]
@manniokim 
Okim, the ohlcdata repository that contained "list.csv" is gone.  Can you please add "list.csv" to the new "bitcoin_analysis" repository?  Thanks. --Daniel

**Comment by manniokim** [comment]
> @manniokim
> Okim, the ohlcdata repository that contained "list.csv" is gone. Can you please add "list.csv" to the new "bitcoin_analysis" repository? Thanks. --Daniel

 so sorry about that, I have just done that

**Comment by DanielGoldfarb** [comment]
@manniokim   ... Thanks

**Comment by Blessvskp** [comment]
The issue raised by @manniokim is extremely relevant and very important to forex, options , commodity and stock traders. 

I am not sure if the issue is completely answered and settled. If settled, I shall be glad to go through the solution.

The possible way is to find the date on which the swing in price is greatest and then by using calendar module, find the day of that date; though I am not completely sure! This requires additional coding.

Dr. Sai
